Include/README.rst

```diff
@@ -0,0 +1,63 @@
+The Python C API
+================
+
+The C API is divided into three sections:
+
+1. ``Include/``
+2. ``Include/cpython/``
+3. ``Include/internal/``
+
+
+Include: Limited API and Stable ABI
+===================================
+
+``Include/`` contains the public Limited API and the Stable ABI.  The
+Stable ABI is defined by :pep:`384`.
+
+Functions stealing references or returning borrowed references *must not*
+be added to the Limited API API or the Stable ABI.  A strong reference
+*must* be returned.  The API in ``Include/`` should not expose or leak
+implementation details.
+
+Please start a public discussion before adding new C API to ``Include/``.
+
+Functions or macros with a ``_Py`` prefix do not belong in ``Include/``.
+
+
+Include/cpython: CPython implementation details
+===============================================
+
+``Include/cpython/`` contains the public API that is excluded from the
+Limited and Stable ABI.  :pep:`384` introduced the ``Py_LIMITED_API``
+define to exclude functions from the Limited API.  When a new API is
+introduced, it has to be explicitly excluded using
+``#ifndef Py_LIMITED_API``.
```

Comment:
It's unclear from this *where* you have to put this. In the header file? And why? If it's under Include/cpython/, isn't it automatically excluded?

Reply:
I was under the impression that parts of a header file are explicitly excluded from the limited API:

```
$ grep -r Py_LIMITED_API Include/{cpython,internal}
Include/cpython/pyarena.h:#ifndef Py_LIMITED_API
Include/cpython/pyarena.h:#endif /* Py_LIMITED_API */
Include/cpython/pyctype.h:#ifndef Py_LIMITED_API
Include/cpython/pyctype.h:#endif /* !Py_LIMITED_API */
Include/cpython/pyfpe.h:#ifndef Py_LIMITED_API
Include/cpython/pyfpe.h:#endif /* !defined(Py_LIMITED_API) */
Include/cpython/odictobject.h:#ifndef Py_LIMITED_API
Include/cpython/picklebufobject.h:#ifndef Py_LIMITED_API
Include/cpython/picklebufobject.h:#endif /* !Py_LIMITED_API */
Include/cpython/pytime.h:#ifndef Py_LIMITED_API
Include/cpython/pytime.h:#endif /* Py_LIMITED_API */
Include/cpython/initconfig.h:#ifndef Py_LIMITED_API
Include/cpython/initconfig.h:#endif /* !Py_LIMITED_API */
Include/cpython/parser_interface.h:#ifndef Py_LIMITED_API
Include/cpython/parser_interface.h:#endif /* !Py_LIMITED_API */
Include/cpython/pydebug.h:#ifndef Py_LIMITED_API
Include/cpython/pydebug.h:#endif /* Py_LIMITED_API */
Include/internal/pycore_accu.h:#ifndef Py_LIMITED_API
Include/internal/pycore_accu.h:#endif /* !Py_LIMITED_API */
Include/internal/pycore_import.h:#ifndef Py_LIMITED_API
Include/internal/pycore_import.h:#endif /* !Py_LIMITED_API */
Include/internal/pycore_bytes_methods.h:#ifndef Py_LIMITED_API
Include/internal/pycore_bytes_methods.h:#endif /* !Py_LIMITED_API */
```

@vstinner, would you mind clarifying?

Reply:
Resolved: https://github.com/python/cpython/pull/24884#discussion_r596444339